core/primitives/src/test_utils.rs

```diff
@@ -569,6 +576,16 @@ impl Block {
             }
         };
     }
+
+    pub fn set_chunk_endorsements(&mut self, chunk_endorsements: Vec<ChunkEndorsementSignatures>) {
+        match self {
+            Block::BlockV1(_) | Block::BlockV2(_) | Block::BlockV3(_) => (),
```

Comment:
Just to double check - we don't need to panic in these cases?

Reply:
If we panic, the tests would fail for older protocol version where we are not dealing with BlockV4.

(I was originally panic-ing here but it didn't work)